src/client/SyntheticClient.cc

```diff
@@ -934,7 +935,9 @@ int SyntheticClient::start_thread()
 
   pthread_create(&thread_id, NULL, synthetic_client_thread_entry, this);
   assert(thread_id);
+#ifdef HAVE_PTHREAD_SETNAME_NP
```

Comment:
could define an empty `pthread_setname_np` in "compat.h" if `HAVE_PTHREAD_SETNAME_NP` is not `defined` like

```c++
#ifndef HAVE_PTHREAD_SETNAME_NP
#define pthread_setname_np(...) 
#endif
```